packages/thirdweb/src/react/web/ui/prebuilt/Chain/name.tsx

```diff
@@ -0,0 +1,185 @@
+   * name was not fetched succesfully
+   * @example
+   * ```tsx
+   * <ChainName fallbackComponent={"Failed to load"}
+   * />
+   * ```
+   */
+  fallbackComponent?: JSX.Element;
+  /**
+   * Optional `useQuery` params
+   */
+  queryOptions?: Omit<UseQueryOptions<string>, "queryFn" | "queryKey">;
+}
+
+/**
+ * This component fetches then shows the name of a chain.
+ * It inherits all the attributes of a HTML <span> component, hence you can style it just like how you would style a normal <span>
+ *
+ *
+ * @example
+ * ### Basic usage
+ * ```tsx
+ * import { ChainProvider, ChainName } from "thirdweb/react";
+ * import { ethereum } from "thirdweb/chains";
+ *
+ * <ChainProvider {...props}>
+ *   <ChainName  />
+ * </ChainProvider>
+ * ```
+ * Result:
+ * ```html
+ * <span>Ethereum Mainnet</span>
+ * ```
+ *
+ * ### Custom name resolver
+ * By default ChainName will call the thirdweb API to retrieve the chain name.
+ * However if you have a different way to fetch the name, you can pass the function to the `nameResolver` prop.
+ * Note: nameResolver should either be a string or a function (async) that returns a string.
+ * ```tsx
+ * async function fetchNameMethod() {
+ *   // your own fetching logic
+ *   return "the chain name";
+ * }
+ *
+ * <ChainName nameResolver={fetchNameMethod} />
+ * ```
+ *
+ * Alternatively you can also pass in a string directly:
+ * ```tsx
+ * <ChainName nameResolver="ETH Mainnet" />
+ * ```
+ *
+ *
+ * ### Format the name (capitalize, truncate, etc.)
+ * The ChainName component accepts a `formatFn` which takes in a string and outputs a string
+ * The function is used to modify the name of the chain
+ *
+ * ```tsx
+ * const concatStr = (str: string):string => str + "Network"
+ *
+ * <ChainProvider {...props}>
+ *   <ChainName formatFn={concatStr} />
+ * </ChainProvider>
+ * ```
+ *
+ * Result:
+ * ```html
+ * <span>Ethereum Mainnet Network</span>
+ * ```
+ *
+ * ### Show a loading sign when the name is being fetched
+ * ```tsx
+ * import { ChainProvider, ChainName } from "thirdweb/react";
+ *
+ * <ChainProvider {...props}>
+ *   <ChainName loadingComponent={<Spinner />} />
+ * </ChainProvider>
+ * ```
+ *
+ * ### Fallback to something when the name fails to resolve
+ * ```tsx
+ * <ChainProvider {...props}>
+ *   <ChainName fallbackComponent={"Failed to load"} />
+ * </ChainProvider>
+ * ```
+ *
+ * ### Custom query options for useQuery
+ * This component uses `@tanstack-query`'s useQuery internally.
+ * You can use the `queryOptions` prop for more fine-grained control
+ * ```tsx
+ * <ChainName
+ *   queryOptions={{
+ *     enabled: isEnabled,
+ *     retry: 4,
+ *   }}
+ * />
+ * ```
+ *
+ * @component
+ * @chain
+ * @beta
+ */
+export function ChainName({
+  nameResolver,
+  formatFn,
+  loadingComponent,
+  fallbackComponent,
+  queryOptions,
+  ...restProps
+}: ChainNameProps) {
+  const { chain } = useChainContext();
+  const nameQuery = useQuery({
```

Comment:
How come we're not using our existing chain hooks here? Otherwise it's not sharing the cache with the rest of the sdk

Reply:
I got a couple of reasons:
- getChainMetadata is already catched (using withCache)
- I need to add extra logic to the queryFn, using useChainMetadata wouldn't give me what I want (and users won't be able to customize it)